data_hub_api/docmaps/provider.py

```diff
@@ -356,14 +357,26 @@ def __init__(
         if only_include_evaluated_preprints:
             self.docmaps_index_query += '\nWHERE has_evaluations\nLIMIT 20'
 
-    def iter_docmaps(self) -> Iterable[dict]:
-        bq_result_iterable = iter_dict_from_bq_query(
-            self.gcp_project_name,
-            self.docmaps_index_query
-        )
+    def get_query_with_doi_where_clause(self, preprint_doi: str) -> str:
+        return self.docmaps_index_query + f'\nAND preprint_doi = {preprint_doi}'
```

Comment:
might be worth adding a note, that this assumes that we added a where clause via `only_include_reviewed_preprint_type` or `only_include_evaluated_preprints`